controllers/database_controller.go

```diff
@@ -166,20 +303,44 @@ func (r *DatabaseReconciler) reconcilePSMDB(ctx context.Context, req ctrl.Reques
 	if err != nil {
 		return err
 	}
-	affinity := &psmdbv1.PodAffinity{
-		TopologyKey: pointer.ToString(psmdbv1.AffinityOff),
-	}
+
+	psmdbSpec := defaultPSMDBSpec
 	if clusterType == ClusterTypeEKS {
-		affinity.TopologyKey = pointer.ToString("kubernetes.io/hostname")
+		affinity := &psmdbv1.PodAffinity{
+			TopologyKey: pointer.ToString("kubernetes.io/hostname"),
+		}
+		psmdbSpec.Replsets[0].MultiAZ.Affinity = affinity
```

Comment:
We need to set affinity off for minikube clusters because they won't work without those rules

Reply:
I don't think I missed them, my intention was to define them as default here...
https://github.com/percona/dbaas-operator/blob/7c309de83d1e07b9237bb84a50d7ed491d93cf51/controllers/database_controller.go#L172-L174
https://github.com/percona/dbaas-operator/blob/7c309de83d1e07b9237bb84a50d7ed491d93cf51/controllers/database_controller.go#L182-L184
https://github.com/percona/dbaas-operator/blob/7c309de83d1e07b9237bb84a50d7ed491d93cf51/controllers/database_controller.go#L189-L191
https://github.com/percona/dbaas-operator/blob/7c309de83d1e07b9237bb84a50d7ed491d93cf51/controllers/database_controller.go#L197-L199
... and only overwrite them if in EKS but please double check if I didn't do any mistake.